Packs/CheckPointSandBlast/Integrations/CheckPointSandBlast/CheckPointSandBlast.py

```diff
@@ -0,0 +1,927 @@
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='SandBlast.Upload',
+        outputs_key_field=['MD5', 'SHA1', 'SHA256'],
+        outputs=output,
+        raw_response=raw_output,
+    )
+
+
+def download_command(client: Client, args: Dict[str, Any]) -> Any:
+    """
+    Download a file from the server.
+    The command will be a bridge between the client request and the presented value to the user.
+    Get arguments inputted by the user and send them to the client.
+    Once a response has been received, process and return it so it can be sent to the user.
+
+    Args:
+        client (Client): Connection to the client class from which we can run the desired request.
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Returns:
+        Any: File from the server.
+    """
+    file_id = args['file_id']
+    output = client.download_request(file_id)
+
+    return fileResult(filename=file_id, data=output)
+
+
+def quota_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Get quota information about an API key.
+    The command will be a bridge between the client request and the presented value to the user.
+    Once a response has been received, process and return it so it can be sent to the user.
+
+    Args:
+        client (Client): Connection to the client class from which we can run the desired request.
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Returns:
+        CommandResults: Quota information about the API key.
+    """
+    raw_outputs = client.quota_request()
+    outputs = raw_outputs.get('response')[0]  # type:ignore
+
+    output = get_quota_context_output(outputs)
+    headers = get_quota_headers()
+
+    readable_output = tableToMarkdown(
+        'Quota Information',
+        output,
+        headers=headers,
+        headerTransform=string_to_table_header,
+        removeNull=True,
+    )
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='SandBlast.Quota',
+        outputs_key_field='QuotaId',
+        outputs=output,
+        raw_response=raw_outputs,
+    )
+
+
+''' POLLING COMMANDS '''
+
+
+def upload_with_polling_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Initiate polling command for upload command.
+
+    Args:
+        client (Client): Connection to the client class from which we can run the desired request.
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Returns:
+        CommandResults: A result to return to the user which will be presented in a markdown value.
+            The result itself will depend on the stage of polling.
+    """
+    return run_polling_command(client, args, 'sandblast-upload', upload_command, query_command)
+
+
+def run_polling_command(
```

Comment:
I've implemented the polling command to use the polling_function decorator, but there is one problem.
The problem is, I can't find a way to pass the user inputs for `interval_in_seconds` and `timeout_in_secods`.
Since the function is built at runtime, I don't think passing the arguments as below will work:

```
interval_in_secs = int(args.get('interval_in_seconds', DEFAULT_INTERVAL))
timeout_in_seconds = int(args.get('timeout_in_seconds', DEFAULT_TIMEOUT))

@polling_function('sandblast-upload', interval=interval_in_seconds, timeout=timeout_in_secods)
def upload_polling_command(args: Dict[str, Any], **kwargs) -> CommandResults:
...
```

A solution to this is running the function without the decorator like this: 

```
interval_in_seconds = arg_to_number(args.get('interval_in_seconds', DEFAULT_INTERVAL))
timeout_in_seconds = arg_to_number(args.get('timeout_in_seconds', DEFAULT_TIMEOUT))

polling_function('sandblast',interval=interval_in_seconds,timeout=timeout_in_seconds)(upload_polling_command)(args, client=client)

def upload_polling_command(args: Dict[str, Any], **kwargs) -> CommandResults:
...
```

As of the current push the user has no way to control the interval and timeout.

Reply:
this should be a workaround:
```
@polling_function(
    name='sandblast-upload',
    interval=arg_to_number(demisto.args().get('polling_interval_in_seconds', DEFAULT_INTERVAL)),
    timeout=arg_to_number(demisto.args().get('polling_timeout', DEFAULT_TIMEOUT)),
)
```

Reply:
Updated!